integration-tests/src/test/java/org/wildfly/prospero/it/commonapi/UpdateTest.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.prospero.it.commonapi;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.deployment.DeployRequest;
+import org.eclipse.aether.deployment.DeploymentException;
+import org.jboss.galleon.ProvisioningException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.wildfly.channel.Channel;
+import org.wildfly.channel.Stream;
+import org.wildfly.prospero.actions.ProvisioningAction;
+import org.wildfly.prospero.actions.UpdateAction;
+import org.wildfly.prospero.api.ProvisioningDefinition;
+import org.wildfly.prospero.cli.CliConsole;
+import org.wildfly.prospero.it.AcceptingConsole;
+import org.wildfly.prospero.model.ChannelRef;
+import org.wildfly.prospero.model.ManifestYamlSupport;
+import org.wildfly.prospero.model.ProsperoConfig;
+import org.wildfly.prospero.model.RepositoryRef;
+import org.wildfly.prospero.test.MetadataTestUtils;
+import org.wildfly.prospero.wfchannel.MavenSessionManager;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+
+public class UpdateTest extends WfCoreTestBase {
+
+    private static final String OUTPUT_DIR = "target/server";
```

Comment:
Just a theoretical note / work for future - I've been thinking that by using this hardcoded "target/server" path we block ourselves from running test cases in parallel (at least I think surefire is able to do that, currently it's disabled). But we use this on plenty of other places still. Alternative is TempFolder.

Reply:
Yes I think we could use a temp folder there.

We might run into some issues with simultaneous maven local-cache updates with parallel tests I think, but it'd be interesting to try